### PR TITLE
ALARMS: Added an alarm listener class.

### PR DESCRIPTION
**Pull request by mattlangford**

### CODE REVIEW

File: command/sub8_alarm/sub8_alarm/alarm_helpers.py [hunk 1]
```diff
@@ -91,3 +91,50 @@ def raise_alarm(self, problem_description=None, parameters=None, severity=None):
...
+            return
+
+        callback = found_alarm['callback']
+        args = found_alarm['args']
+
+        #Actaully call the callback function and pass args if nessicary.
+        if args is None:
+            callback(alarm)
+        else:
+            callback(alarm, args)
```
Comment (jpanikulam) [inline command/sub8_alarm/sub8_alarm/alarm_helpers.py:140]: Use *args so the args can be passed directly

**Comment by jpanikulam** [comment]
Matt: great stuff, I love the comments. 
- In the future, if you pull --rebase you can avoid having to merge. But sometimes you have to merge and that's ok. 
- run flake8 --max-line-length=130 on this file, there are some minor violations

**Comment by jpanikulam** [comment]
This closes #77 
@zachgoins

File: command/sub8_alarm/sub8_alarm/alarm_helpers.py (hunk 1 above)
Comment (mattlangford) [inline command/sub8_alarm/sub8_alarm/alarm_helpers.py:140]: Just to confirm before I recommit - if there are multiple args, they still have to be passed to the listener as a list but they will be sent out as multiple arguments to the function callback.

**Comment by zachgoins** [comment]
@mattlangford @jpanikulam you guys da best

File: command/sub8_alarm/sub8_alarm/alarm_helpers.py (hunk 1 above)
Comment (jpanikulam) [inline command/sub8_alarm/sub8_alarm/alarm_helpers.py:140]: Yep.
 If you have add_listener take a *args, then you can pass a bunch of arguments in the same way that you send them to the callback.

**Comment by jpanikulam** [comment]
Merging now so Zach can use - address minor stuff in a future commit

**Comment by forrestv** [comment]
Seven minutes too late, but:

There's really no reason to have an explicit `args`/`*args` argument. I know Twisted has a lot of that kind of thing in their API, but I think that that's just because Twisted started before the `lambda` keyword was introduced to Python.

With `lambda`, you can create your own little anonymous function to add the extra arguments if you need to, without adding complexity to the API for everyone who doesn't need to. To the user it'd be something like:

```
alarm_listener.add_listener(alarm_name, lambda alarm: my_real_callback(alarm, 1, 2, 3))
```

rather than

```
alarm_listener.add_listener(alarm_name, my_real_callback, 1, 2, 3)
```

This simplifies AlarmListener a lot too.

**Comment by mattlangford** [comment]
That's an interesting way to do it. I didn't know you could pass lamdas around like that though I shouldn't be surprised. I was modeling the callback like a ros subscriber where you can pass optional arguments in, but this definitely simplifies it a lot - I will make this change for my next commit.

**Pull request merged**